### PR TITLE
Create a terraform module for Tempo

### PR DESCRIPTION
**Pull request by michaeldmitry**
Add a terraform module for Tempo coordinator. 

## Context
https://github.com/canonical/mimir-worker-k8s-operator/pull/74
https://docs.google.com/document/d/1EG71A2pJ244PQRaGVzGj7Mx2B_bzE4U_OSqx4eeVI1E/edit

## TODO
- Automated tests will be part of a split task

## Testing Considerations
Install terraform on your machine
`cd terraform/ && terrform init && terraform apply`

### CODE REVIEW

File: .github/workflows/pull-request.yaml [hunk 1]
```diff
@@ -10,3 +10,7 @@ jobs:
     name: PR
     uses: canonical/observability/.github/workflows/charm-pull-request.yaml@main
     secrets: inherit
+
+  terraform-checks:
+    name: Terraform
+    uses: canonical/observability/.github/workflows/terraform-quality-checks.yaml@OPENG-2685
```
Comment (mmkay) [inline .github/workflows/pull-request.yaml:16]: Remember to replace it with main at merging (I guess you'll be merging all three PRs at the same time) :)

File: terraform/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+| Name | Description |
+| - | - |
+| `app_name`|  Application name |
+| `provides`| Map of `provides` endpoints |
+| `requires`|  Map of `requires` endpoints |
+
+## Usage
+
+TODO: Update Tempo HA Terraform module link
```
Comment (mmkay) [inline terraform/README.md:34]: Guessing you can do that already?

File: terraform/variables.tf [hunk 3]
```diff
@@ -0,0 +1,35 @@
+variable "app_name" {
+  description = "Application name"
+  type        = string
+  default     = "tempo"
+}
+
+variable "channel" {
+  description = "Charm channel"
+  type        = string
+  default     = "latest/stable"
```
Comment (mmkay) [inline terraform/variables.tf:10]: Same comment as in https://github.com/canonical/observability/pull/209#discussion_r1782428694 : maybe it's worth a follow-up PR once we release to stable?

**Review by mmkay — APPROVED** [review]

**Comment by michaeldmitry** [comment]
Itests fail due to an unrelated issue to this PR changes. Will be addressed in a separate PR

**Pull request merged**